lib/experimental/blocks.php

```diff
@@ -100,3 +100,72 @@ function gutenberg_register_metadata_attribute( $args ) {
 	return $args;
 }
 add_filter( 'register_block_type_args', 'gutenberg_register_metadata_attribute' );
+
+/**
+ * Filters the block being rendered in render_block(), before it's processed.
+ *
+ * @param array         $parsed_block The block being rendered.
+ * @param array         $source_block An un-modified copy of $parsed_block, as it appeared in the source content.
+ * @param WP_Block|null $parent_block If this is a nested block, a reference to the parent block.
+ */
+function gutenberg_auto_insert_child_block( $parsed_block, $source_block, $parent_block ) {
+	if ( isset( $parent_block ) ) {
+		$parsed_block['parentBlock'] = $parent_block->name;
+	}
+	return $parsed_block;
+}
+add_filter( 'render_block_data', 'gutenberg_auto_insert_child_block', 10, 3 );
+
+/**
+ * Auto-insert blocks relative to a given block.
+ *
+ * @param string   $block_content The block content.
+ * @param array    $block         The full block, including name and attributes.
+ */
+function gutenberg_auto_insert_blocks( $block_content, $block ) {
+	// $block_name     = 'core/post-content';
+	// $block_position = 'after'; // Child blocks could be a bit trickier.
+
+	$block_name     = 'core/comment-template';
+	$block_position = 'last-child';
+
+	// Can we void infinite loops?
+
+	if (
+		! (
+			$block_name === $block['blockName'] &&
+			( 'before' === $block_position || 'after' === $block_position )
+		) && ! (
+			isset( $block['parentBlock'] ) &&
+			$block_name === $block['parentBlock'] &&
+			( 'first-child' === $block_position || 'last-child' === $block_position )
+		)
+	) {
+		return $block_content;
+	}
+
+	$inserted_block_markup = '<!-- wp:avatar {"size":40,"style":{"border":{"radius":"10px"}}} /-->';
+
+	$inserted_blocks  = parse_blocks( $inserted_block_markup );
+	$inserted_content = render_block( $inserted_blocks[0] );
```

Comment:
It seems like we could replace these three lines with:

```php
$inserted_content = render_block( array(
	'blockName'    => 'core/avatar',
	'attrs'               => array(
		'size' => 40,
		'style => array(
			'border' => array( 'radius' => '10px' ),
		),
	),
	'innerHTML'    => '',
	'innerContent' => array(),
) );
```

Reply:
Applied in https://github.com/WordPress/gutenberg/pull/50103/commits/da7290d31b49be96f38a797064339875ddb7ec57 😊